lib/core/test/unit/Cardano/WalletSpec.hs

```diff
@@ -431,7 +431,10 @@ dummyTransactionLayer = TransactionLayer
             let sig = CC.unXSignature $ CC.sign pwd (getKey xprv) sigData
             return $ TxWitness
                 (CC.unXPub (getKey $ publicKey xprv) <> sig)
-        return (tx, wit)
+
+        -- (tx1, wit1) == (tx2, wit2) <==> fakebinary1 == fakebinary2
```

Comment:
1. I presume is what we're interested in in the above tests
2. Could probably add this as a property (or better, something with the test that is requiring `mkStdTx` 🤔, is there really one? 🤔  )
3. This use-case makes me _slightly_ wish that `SignedTxBinary` was some kind of `ToBinary tx => tx`, but not enough to make me doubt `SignedTxBinary` yet.

Reply:
` was some kind of ToBinary tx => tx` 

No. We are actually doing more than just a binary serialization here. A signed transaction has a cryptographic signature that is produced not only from the context of a mere `tx`.

Reply:
1. Correct.
2. It is a property already. That's what `walletKeyIsReencrypted` is checking. But Not with the actual transaction layer though I give you that.

Reply:
>  That's what walletKeyIsReencrypted is checking.

Not quite, right? I was thinking we should have a test that would prevent others from (like me) sabotaging the effectiveness of `walletKeyIsReencrypted `.

But… I'd happy happy to not to 🤷‍♂

Reply:
Such test is a good idea. But not in this PR. Keep PR manageable and avoid increasing their scope ad infinitum.

If need be, add a point to the original ticket.